audpsychometric/core/gold_standard.py

```diff
@@ -1,122 +1,113 @@
-r"""Calculating gold standards over individual raters' judgments."""
-
+import typing
 
 import numpy as np
 import pandas as pd
 
 import audmetric
 
 
-def gold_standard_mean(
-    df: pd.DataFrame,
-    minimum: float,
-    maximum: float,
+def confidence_categorical(
+    ratings: typing.Sequence,
     *,
     axis: int = 1,
-) -> pd.DataFrame:
-    r"""Calculate the gold standard as the mean of raters' votes.
-
-    This functional uses the numerical confidence calculation.
+) -> typing.Union[float, np.ndarray]:
+    r"""Confidence score for categorical ratings.
 
-    The returned table
-    has an index identical to the input :class:`pd.DataFrame`,
-    and additional columns `gold_standard` and `confidence`.
+    The confidence for categorical data
+    is given by the fraction of raters per item
+    with the rating being equal to that of the gold standard
+    as given by :func:`audpsychometric.mode`.
```

Comment:
No, `audpsychometric.mode()` will always convert to the next integer and does not support floats:
```python
>>> audpsychometric.mode([0, 1])
1
>>> audpsychometric.mode([0, 2])
1
```
As this is different from the usual mode implementation, I wanted to mention that we use our implementation here.
So maybe I change to
```suggestion
    with the rating being equal to the mode
    given by :func:`audpsychometric.mode`.
```
?

I have also updated the docstring of `mode()` to mention the rounding.

Reply:
In general, our `mode()` implementation is very special, as it assumes for integer classes that they are ordered and have no missing classes, which is a strong assumption. But if met, it provides better results than the normal `mode`. I first thought about adding an argument to disable the behavior, but then a user can also just use the standard `mode` implementation of another package.

Reply:
I see, thanks for clarifying! I agree with the suggestion.

I think the implementation for integer classes makes sense as it is.

The only additional feature I would like, would be some more options for the non-numerical case, when there are multiple winning categories (return a user-specified value, e.g. "no agreement", or return a random winning value rather than the first occurring one). But I was going to add an issue after this PR.

Reply:
Yes, please open an issue to extend the functionality for non-numerical cases. We would need to add the argument then to `mode()` and to `agreement_categorical()` as this uses `mode()`.